app/models/Quarters.scala

```diff
@@ -69,12 +76,37 @@ object Quarters {
       case _ => Q4
     }
 
-}
+  def validQuartersForYear(year: Int)(implicit config: FrontendAppConfig): Seq[Quarters] = {
+    val earliestValidYear: Int = LocalDate.parse(config.quarterStartDate).getYear
+    year match {
+      case _ if year == currentYear => getCurrentYearQuarters
```

Comment:
Just say case currentYear =>

Reply:
scala does not recognise it as an existing val because cases need to predefined static values.